src/yvalve/PluginManager.cpp

```diff
@@ -605,7 +605,7 @@ namespace
 		FbLocalStatus ls;
 		IPluginBase* plugin = module->getPlugin(regPlugin).factory->createPlugin(&ls, par);
 
-		if (!(ls->getState() & Firebird::IStatus::STATE_ERRORS))
+		if (plugin && !(ls->getState() & Firebird::IStatus::STATE_ERRORS))
```

Comment:
Why `createPlugin()` may return `nullptr` with successful status? This smells like a bug.

Reply:
Yes, it is a bug. But crash isn't an adequate reaction from Firebird to such bug.

Reply:
Besides API specs doesn't mention that nullptr is not allowed to be returned from `PluginFactory::createPlugin()` so it is legit for the factory to refuse to provide plugin instance because of given configuration or just out of blue.

Reply:
If you want to speak about correct (legal) implementation: plugin factory should set error if it doesn't creates plugin by any reason.
Every method which have `IStatus` as parameter must use it to report errors.

Anyway, I agree that caller should check returned pointer for `nullptr`.
Also, will be good to log message if plugin factory returns `nullptr` and set no error, IMO.
It helps plugin developer to see the bug.